nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/GetSlackReaction.java

```diff
@@ -0,0 +1,524 @@
+            Set 'Message Identifier Strategy' to 'JSON Path' (channel ID and message timestamp will be taken from the JSON output of ConsumeSlack/ListenSlack)
+            You can leave "Channel ID JSON Path", "Message Timestamp JSON Path" and "Message Text JSON Path" on default values.
+            Set "Release Per Reaction" to 'true' if you wait for the first reaction(s) only, in case want to collect all reactions in a given waiting period,
+            set this property to 'false'.
+            Set "Wait Period" to the time period the processor will continue to search for reactions for the specified message (unless releasing per reaction)
+            """
+)
+public class GetSlackReaction extends AbstractProcessor {
+    public static final String ATTR_MESSAGE_TIMESTAMP = "slack.ts";
+    public static final String ATTR_CHANNEL_ID = "slack.channel.id";
+    public static final String ATTR_ERROR_MESSAGE = "error.message";
+    public static final String ATTR_FIRST_REACTION_CHECK_TIMESTAMP = "first.reaction.check.ts";
+    public static final String ATTR_LAST_REACTION_CHECK_TIMESTAMP = "last.reaction.check.ts";
+    public static final String ATTR_REACTION_POLLING_SECONDS = "reaction.polling.seconds";
+    public static final String ATTR_SLACK_MESSAGE_COUNT = "slack.message.count";
+    public static final int MIN_REACTION_POLLING_FREQUENCY_SECONDS = 10;
+
+
+    static final PropertyDescriptor ACCESS_TOKEN = new PropertyDescriptor.Builder()
+            .name("Access Token")
+            .description("OAuth Access Token used for authenticating/authorizing the Slack request sent by NiFi. This may be either a User Token or a Bot Token. " +
+                    "It must be granted the reactions:read scope.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .required(true)
+            .sensitive(true)
+            .build();
+
+    static final PropertyDescriptor MESSAGE_IDENTIFIER_STRATEGY = new PropertyDescriptor.Builder()
+            .name("Message Identifier Strategy")
+            .description("Specifies the strategy to obtain the message timestamp and channel id from incoming Flow File. "
+             + "In case of '" + MessageIdentifierStrategy.ATTRIBUTES.getValue() + "' the 'slack.channel.id' and 'slack.ts' Flow File attributes will be used "
+             + "and in case of '" + MessageIdentifierStrategy.JSON_PATH.getValue() + "' additional JSON path values needs to be specified to fetch channel id "
+             + "and message timestamp from incoming JSON content.")
+            .required(true)
+            .allowableValues(MessageIdentifierStrategy.class)
+            .defaultValue(MessageIdentifierStrategy.ATTRIBUTES.getValue())
+            .build();
+
+    static final PropertyDescriptor CHANNEL_ID_JSON_PATH = new PropertyDescriptor.Builder()
+            .name("Channel ID JSON Path")
+            .description("The JSON Path which identifies the channel ID in the incoming JSON.")
```

Comment:
Should we require input to be in JSON format? It might be more flexible (and facilitate a future ConsumeSlackRecord processor for example) to use Record Path and a Record Reader?